src/ucp/api/ucp.h

```diff
@@ -1665,6 +1665,23 @@ ucs_status_t ucp_listener_reject(ucp_listener_h listener,
                                  ucp_conn_request_h conn_request);
 
 
+/**
+ * @ingroup UCP_WORKER
+ *
+ * @brief Get the port number on which the listener is listening on.
+ *
+ * Get the port number on which the given listener is listening for incoming
+ * connection requests.
+ * This function should be called only after the call to ucp_listener_create()
+ * returned successfully.
+ *
+ * @param [in]  listener        Handle to the listener.
+ * @param [out] conn_request    The port the listener is listening on for incoming
+ *                              connection requests.
+ */
+void ucp_listener_get_listening_port(ucp_listener_h listener, int *listen_port_p);
```

Comment:
maybe extend to `uct_status_t ucp_listener_query(ucp_listener_h, ucp_listener_attr *)`? and the same for UCT, btw it could be iface_attr